R/gsmApp_Server.R

```diff
@@ -108,8 +111,20 @@ gsmApp_Server <- function(
     )
 
     ## Metric Details ----
-    srvr_SyncTab("primary_nav_bar", "Metric Details", rctv_InputMetric, session)
-    srvr_SyncTab("primary_nav_bar", "Metric Details", rctv_InputSite, session)
+    srvr_SyncTab(
```

Comment:
Does this tie an update to an input to navigation?

Reply:
Yup! I tweaked it to only navigate if they're on one of the main tabs, because it was annoying to move while interacting with the plugin. We can tweak it further based on user feedback.

This is the function:

```r
srvr_SyncTab <- function(
  id,
  strTargetTab,
  rctv_strValue,
  rctv_strCurrentTab,
  session = getDefaultReactiveDomain()
) {
  observe({
    main_tabs <- c("Study Overview", "Metric Details", "Participant Details")
    if (rctv_strCurrentTab() %in% main_tabs) {
      bslib::nav_select(id, strTargetTab, session = session)
    }
  }) %>%
    bindEvent(
      null_for_none(rctv_strValue()),
      ignoreInit = TRUE
    )
}
```